Sources/DataPipeline/DataPipelineImplementation.swift

```diff
@@ -96,6 +119,39 @@ class DataPipelineImplementation: DataPipelineInstance {
         }
     }
 
+    /// Adds device default and custom attributes using DeviceAttributes plugin
+    private func addDeviceAttributes(_ customAttributes: [String: Any]) {
+        // OS name might not be available if running on non-apple product. We currently only support iOS for the SDK
+        // and iOS should always be non-nil. Though, we are consolidating all Apple platforms under iOS but this check
+        // is
+        // required to prevent SDK execution for unsupported OS.
+        if deviceInfo.osName == nil {
+            logger.info("SDK being executed from unsupported OS. Ignoring request to register push token.")
+            return
+        }
+
+        // Consolidate all Apple platforms under iOS
+        let deviceOsName = "iOS"
+        // FIXME: [CDP] Fetch the right defaultDeviceAttributes here
+        // deviceAttributesProvider.getDefaultDeviceAttributes { defaultDeviceAttributes in
+        let defaultDeviceAttributes: [String: Any] = [:]
```

Comment:
`DeviceAttributesProvider` relies on `SdkConfig` for SDK version and wrapper config. I'll refactor it in upcoming PRs and fix this.

Reply:
Fixed ✅